packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -123,15 +125,49 @@ export async function executePlugins(
 
     progressBar?.updateTitle(`Executing  ${chalk.bold(pluginCfg.title)}`);
 
-    const pluginReport = await executePlugin(pluginCfg);
-    progressBar?.incrementInSteps(plugins.length);
-
-    return outputs.concat(pluginReport);
-  }, Promise.resolve([] as PluginReport[]));
+    try {
+      const pluginReport = await executePlugin(pluginCfg);
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat(Promise.resolve(pluginReport));
+    } catch (e: unknown) {
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat(
+        Promise.reject(e instanceof Error ? e.message : String(e)),
+      );
+    }
+  }, Promise.resolve([] as Promise<ExecutePluginResult>[]));
 
   progressBar?.endProgress('Done running plugins');
 
-  return pluginsResult;
+  const errorsCallback = ({ reason }: PromiseRejectedResult) =>
+    console.error(reason);
+
+  return Promise.allSettled(pluginsResult).then(results => {
+    logMultipleResults(results, 'Plugins', undefined, errorsCallback);
+
+    const failedResults = results.filter(
```

Comment:
We need something like groupBy here.

```
const {fulfilled, rejected} = groupBy('status', results);
```

I can be more readable and less complex.

Reply:
We can move it into a new issue?

here a PoC:

```typescript
function groupBy<T extends Record<Key, any>>(key: Key, array: T[]): Record<string, T[]> {
  return array.reduce((accumulator, currentValue) => {
    const groupKey = currentValue[key];
    if (!accumulator[groupKey]) {
      accumulator[groupKey] = [];
    }
    accumulator[groupKey].push(currentValue);
    return accumulator;
  }, {} as Record<string, T[]>);
}
```

Reply:
New ticket #287 was created, will be in that scope.